platforms/chibios/common/ld/STM32F103x8_stm32duino_bootloader.ld

```diff
@@ -19,7 +19,7 @@
  */
 MEMORY
 {
-    flash0  : org = 0x08002000, len = 64k - 0x2000
+    flash0  : org = 0x08002000, len = 128k - 0x2000
```

Comment:
This change should be reverted.  Or part of a separate PR

Reply:
Right, so I was kind of afraid of that. I would guess you are aware of the STM32F103C8T6 "feature" where it is supposed to have 64k of flash, but actually comes with 128k. 

This is known to QMK as a whole because the EEPROM emulation on that exact MCU is placed at the top of 128k. 

The reason I have it being committed here is because I am barely over the 64k - 0x2000 limit. I don't think I am actually over 64k, but within 0x2000 anyway. What should I do? This works, I've already tested it on a couple of boards no problem. I can commit it as it's own commit if you want, but since it is technically not a 128k part I wasn't sure what y'all would think.

Reply:
No, I wasn't really aware of it, though, I vaguely remember seeing something about it on discord. 

However, We generally try to keep core and keyboard PRs separate. 

I would prefer that it be a separate PR, and either wait on this being merged, or maybe see about using the nano spec, I think it was. Or if a custom ld can be used here? 

But I'll defer to @tzarc on this, I think.

Reply:
Great, thanks. If there was some way to override the size from within the keyboard folder that'd be ideal. I'll wait for recommendations from @tzarc.

Reply:
Note that clone chips like CKS32F103C8T6, which are often found on Bluepill boards from AliExpress, don't have the extra 64K flash: #7204. I guess this is not a problem when you are making your own PCB and can control what components are actually used.

In the longer term this should be solved with something like #8179.

Reply:
I agree, in the case of clone chips this would be an issue. But to be clear, the EEPROM emulation would also be an issue since it is done on the assumption that 128k is available. Perhaps there could be a separate setting for clones? Or the flash suffix would also work of course. If there is currently no way to make this flash size change in a keyboard specific way (which it sounds like there isn't) then I can go ahead and remove that change and the PR can proceed. The only issue being that I would need to update the README or something to specify that for the time being that file has to be changed before it will build properly.

Reply:
I also noticed in the `STM32F103x8_stm32duino_bootloader.ld` file 0x2000 or 8kb of flash is set aside (for EEPROM presumably), isn't that a ton? It looks like from `eeprom_stm32.h` that only 0x800 or 2kb is used. See line 46 and 47. So an additional 6kb of space could be freed up without issue by changing the 0x2000 to 0x800. Correct? If so I have no issue reverting the 128k change and making a second PR to fix this as that should allow plenty of space for my changes. Win-win.

Reply:
`STM32F103x8_stm32duino_bootloader.ld ` actually reserves 8K of flash for the bootloader (currently the bootloader binaries are around 7K), and does not reserve anything for the EEPROM (which is another problem that should really be considered, and it's actually a concern for many other chips, not just F103).

In theory you can add the custom ld file to your board files and specify it in `MCU_LDSCRIPT`, but you would also need to remove `BOOTLOADER = stm32duino` and copy the rest of setup which is normally done by `bootloader.mk` to your `rules.mk`:
```make
OPT_DEFS += -DBOOTLOADER_STM32DUINO
BOARD = STM32_F103_STM32DUINO
STM32_BOOTLOADER_ADDRESS = 0x80000000
DFU_ARGS = -d 1EAF:0003 -a 2 -R
DFU_SUFFIX_ARGS = -v 1EAF -p 0003
```
(You cannot rely on the code from `bootloader.mk`, because it overrides `MCU_LDSCRIPT` unconditionally, so that it would also override the default setting from `quantum/mcu_selection.mk`.)

Obviously such code copying is not really desired, and the proper solution would probably involve something like #8179 with subsequent makefile modifications to select the ld file and other options according to the specified MCU flash size.